### PR TITLE
build.yaml: Fix FROM_FORK value, ReleaseOS PR builds

### PR DESCRIPTION
**Pull request by bennettgoble**
Fix up builds from PRs by adding the missing `FROM_FORK` environment variable which must have gone missing during a rebase.

### CODE REVIEW

**Comment by bennettgoble** [comment]
Looks like macOS ReleaseOS may be broken? The ostensibly open source build is attempting to install a proprietary library, havok-source.

**Review by akleshchev — APPROVED** [review]

**Comment by akleshchev** [comment]
For some reason -DHAVOK:BOOL=ON' is set, but shouldn't be.

> '-DCMAKE_BUILD_TYPE:STRING=Release'

Shouldn't this be ReleaseOS?

**Comment by marchcat** [comment]
It's being set at [build.sh#L150](https://github.com/secondlife/viewer/blame/5809b391434245e9c15688a6d4678a827eff72df/build.sh#L150). However, I don't see any code there to disable it. How do we build ReleaseOS on windows?"

**Comment by akleshchev** [comment]
That `HAVOK=ON` does look like it should be conditional instead of always ON.

**Comment by marchcat** [comment]
That's right. Probably something like this at [build.sh#L152](https://github.com/secondlife/viewer/blob/5809b391434245e9c15688a6d4678a827eff72df/build.sh#L152)
```
if [[ "$arch" == "Darwin" ]]
then
    if [[ "$variant" == "Release" ]]
    then
        SIGNING=("-DENABLE_SIGNING:BOOL=YES" \
                 "-DSIGNING_IDENTITY:STRING=Developer ID Application: Linden Research, Inc.")
    else
        # disable havok for non-release builds
        HAVOK=OFF
    fi
fi

```

File: build.sh [hunk 1]
```diff
@@ -146,12 +146,21 @@ pre_build()
     && [ -r "$master_message_template_checkout/message_template.msg" ] \
     && template_verifier_master_url="-DTEMPLATE_VERIFIER_MASTER_URL=file://$master_message_template_checkout/message_template.msg"
 
-    RELEASE_CRASH_REPORTING=ON
-    HAVOK=ON
+    RELEASE_CRASH_REPORTING=OFF
+    HAVOK=OFF
```
Comment (marchcat) [inline build.sh:150]: That's even better than I thought.

File: build.sh [hunk 2]
```diff
@@ -146,12 +146,21 @@ pre_build()
...
-                  "-DSIGNING_IDENTITY:STRING=Developer ID Application: Linden Research, Inc.")
+    if [[ "$variant" == "Release" ]]
+    then
+        RELEASE_CRASH_REPORTING=ON
+        HAVOK=ON
+
+        if [[ "$arch" == "Darwin" ]]
+        then
+            RELEASE_CRASH_REPORTING=ON
+            HAVOK=ON
```
Comment (marchcat) [inline build.sh:160]: These two look redundant here, but it's fine.

**Review by marchcat — APPROVED** [review]
I see `'-DHAVOK:BOOL=OFF'` in the log, so the fix works.

File: build.sh (hunk 2 above)
Comment (bennettgoble) [reply build.sh:160]: Whoops. Fixed, thanks.

**Review by akleshchev — APPROVED** [review]

**Pull request merged**